src/app/features/ControlPanel/ControlPanelController.js

```diff
@@ -12,14 +12,20 @@ require('./Services/ServerStatusService');
 
 var ControlPanelController = (function () {
     var ControlPanelController = function ControlPanelController (
-        runnerService,
+        runnerService,
         serverStatusService,
-        config
+        config,
+        $scope
     ) {
         this.runnerService = runnerService;
         this.serverStatusService = serverStatusService;
-        this.environments = config.environments;
-        this.environment = _.first(this.environments);
+        this.environments = config.environments;
+        this.environment = _.first(this.environments);       
+        $scope.$watch('controlPanel.environment', function (newEnvironmentValue, oldEnvironmentValue) {

```

Comment:
Following from my notes on the `RunnerService` here you would use another `get` and `set` on `this.environment` that updates the value within `this.runnerService`.